File: tests/benchmark/fibonacci/ocaml/fibonacci.ml
(* compute the Nth Fibonacci number modulo 2^28 *)

let fib k =
  let rec go k n m =
    if k == 0 then
      n
    else
      go (k - 1) m ((n + m) mod 268435456)
  in
  go k 0 1

;;

print_int (fib 100000000);
print_newline ()

File: tests/benchmark/fibonacci/c/fibonacci.c
// Compute the Nth Fibonacci number modulo 2 ^ 28

#include <stdio.h>

static int fib(int k) {
    int n = 0;
    int m = 1;
    for (int i = 0; i < k; ++i) {
        int x = (n + m) % (1 << 28);
        n = m;
        m = x;
    }
    return n;
}

int main() {
    printf("%d\n", fib(1000000));
    return 0;
}
